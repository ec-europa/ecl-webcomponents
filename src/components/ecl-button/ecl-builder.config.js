const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-button-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-button-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-button-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-button-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
