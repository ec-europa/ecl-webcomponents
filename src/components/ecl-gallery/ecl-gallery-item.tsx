import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'ecl-gallery-item',
})

export class EclGalleryItem {
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() imageAlt: string;
  @Prop() mediaHref: string;
  @Prop() mediaIframeHref: string;
  @Prop() mediaSharePath: string;
  @Prop() meta: string;
  @Prop() type: string = 'image';
  @Prop() icon: string = 'video';

  getClass(): string {
    return [
      `ecl-gallery__item`,
      `sc-ecl-gallery-${this.theme}`,
      this.styleClass
    ].join(' ');
  }

  getLinkAttr() {
    const attrs = {
      'data-ecl-gallery-item': ''
    };

    if (this.mediaSharePath) {
      attrs['data-ecl-gallery-item-share'] = this.mediaSharePath;
    }
    if (this.mediaIframeHref) {
      attrs['data-ecl-gallery-item-embed-src'] = this.mediaIframeHref;
    }

    return attrs;
  } 

  render() {
    return (
      <li class={this.getClass()}>
        <a
          href={this.mediaIframeHref ? this.mediaIframeHref : this.mediaHref}
          class={`ecl-gallery__item-link sc-ecl-gallery-${this.theme}`}
          {...this.getLinkAttr()}
        >
          <figure class={`ecl-gallery__image-container sc-ecl-gallery-${this.theme}`}>
            <slot name="video"></slot>
          { this.type !== 'html-video' ?
            <img 
              src={this.mediaHref}
              alt={this.imageAlt}
              class={`ecl-gallery__image sc-ecl-gallery-${this.theme}`}
            /> : ''
          }
          { this.type !== 'image' ?
            <ecl-icon
              style-class={`ecl-gallery__image-icon sc-ecl-gallery-${this.theme}`}
              icon={this.icon}
              size="l"
            ></ecl-icon> : ''
          }
            <figcaption
              class={`ecl-gallery__description sc-ecl-gallery-${this.theme}`}
              data-ecl-gallery-description
            >
              <slot></slot>
              <span
                class={`ecl-gallery__meta sc-ecl-gallery-${this.theme}`}
                data-ecl-gallery-meta
              >
                {this.meta}
              </span>
            </figcaption>
          </figure>
        </a>
      </li>
    );
  }
}
