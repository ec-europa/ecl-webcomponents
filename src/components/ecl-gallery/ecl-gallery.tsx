import { Component, h, Prop, Element } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";
declare const ECL: any;

@Component({
  tag: 'ecl-gallery',
  styleUrls: {
    ec: './build/styles/ecl-gallery-ec.css',
    eu: './build/styles/ecl-gallery-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclGallery {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() eclScript: boolean = false;
  @Prop() slidesNumber: number;
  @Prop() counterLabel: string ;
  @Prop() counterSeparator: string;
  @Prop() description: string;
  @Prop() meta: string;
  @Prop() nextLabel: string;
  @Prop() prevLabel: string;
  @Prop() closeLabel: string;
  @Prop() shareLabel: string;
  @Prop() footerLinkPath: string;
  @Prop() footerLinkLabel: string;
  @Prop() visibleItems: number = 8;
  @Prop() fullWidth: boolean = false;
  @Prop() viewAllLabel: string;
  @Prop() viewAllExpandedLabel: string;
  @Prop() expandable: boolean = true;
  @Prop() fullScreenLabel: string = '';
  @Prop() noOverlay: boolean = false;
  @Prop() ariaLabel: string;

  getClass(): string {
    const styleClasses = [
      `ecl-gallery`,
      this.styleClass
    ];

    if (this.fullWidth) {
      styleClasses.push('ecl-gallery--full-width');
    }

    return styleClasses.join(' ');
  }

  componentDidRender() {
    const items = this.el.querySelectorAll('.ecl-gallery__item');
    this.el.querySelector('.ecl-gallery__list').innerHTML = '';
    this.el.querySelector('.ecl-gallery__list').append(...items);

    if (this.eclScript) {
      const src = getAssetPath('./build/scripts/ecl-gallery-vanilla.js');
      if (document.querySelector(`script[src="${src}"]`)) {
        document.querySelector(`script[src="${src}"]`).remove();
      }
      const script = document.createElement('script');
      script.src = src;
      script.onload = () => {
        const gallery = new ECL.Gallery(this.el.firstElementChild);
        gallery.init();
      };
      document.body.appendChild(script);
    }
  }

  render() {
    return (
      <section
        class={this.getClass()}
        data-ecl-gallery
        data-ecl-gallery-visible-items={this.visibleItems}
        {...!this.expandable && ({ 'data-ecl-gallery-not-expandable' : true })}
        {...this.noOverlay && ({ 'data-ecl-gallery-no-overlay' : true })}
      >
        <ul class="ecl-gallery__list">
          <slot></slot>
        </ul>
        <div class="ecl-gallery__footer">
        {this.expandable ? (
          <div class="ecl-gallery__info">
            <strong class="ecl-gallery__info-total" data-ecl-gallery-count>0</strong>
            {this.counterLabel},
            {this.viewAllLabel || this.viewAllExpandedLabel ? (
              <ecl-button
                styleClass={`ecl-gallery__view-all sc-ecl-gallery-${this.theme}`}
                variant="ghost"
                data-ecl-gallery-collapsed-label={this.viewAllLabel}
                data-ecl-gallery-expanded-label={this.viewAllExpandedLabel}
                data-ecl-gallery-all
              >
                {this.viewAllLabel}
              </ecl-button>
            ) : null}
          </div>
        ) : null}
          <hr class="ecl-gallery__footer-divider" />
        {this.footerLinkPath && this.footerLinkLabel ? (
          <div class="ecl-gallery__footer-link">
            <ecl-link
              path={this.footerLinkPath}
              variant="standalone"
              styleClass={`sc-ecl-gallery-${this.theme}`}
            >
              {this.footerLinkLabel} 
            </ecl-link>
          </div>
          ) : null
        }
        </div>
        <dialog
          class="ecl-gallery__overlay"
          data-ecl-gallery-overlay
          aria-label={this.ariaLabel || null}
        >
          <header
            class="ecl-gallery__close"
            data-ecl-gallery-overlay-header
          >
            <ecl-button
              theme={this.theme}
              style-class={`ecl-gallery__close-button sc-ecl-gallery-${this.theme}`}
              variant="ghost"
              data-ecl-gallery-close
            >
              {this.closeLabel}
              <ecl-icon
                theme={this.theme}
                style-class={`sc-ecl-gallery-${this.theme}`}
                icon="close-filled"
                size="s"
                slot="icon-after"
              ></ecl-icon>
            </ecl-button>
          </header>
          <section class="ecl-gallery__slider">
            <div
              class="ecl-gallery__slider-media-container"
              data-ecl-gallery-overlay-media
            ></div>
          </section>
          <footer
            class="ecl-gallery__detail"
            data-ecl-gallery-overlay-footer
          >
            <div class="ecl-container">
             <div class="ecl-gallery__detail-actions">
                <ecl-link
                  theme={this.theme}
                  style-class={`ecl-gallery__download sc-ecl-gallery-${this.theme}`}
                  data-ecl-gallery-overlay-download
                  variant="standalone"
                  path=""
                  target="blank"
                >
                  {this.fullScreenLabel}
                  <ecl-icon
                    theme={this.theme}
                    icon="fullscreen"
                    size="fluid"
                    slot="icon-after"
                  ></ecl-icon>

                </ecl-link>
              { this.shareLabel ? 
                <ecl-link
                  theme={this.theme}
                  style-class={`ecl-gallery__share sc-ecl-gallery-${this.theme}`}
                  data-ecl-gallery-overlay-share
                  variant="standalone"
                  path=""
                >
                  {this.shareLabel}
                  <ecl-icon
                    theme={this.theme}
                    icon="share"
                    size="fluid"
                    slot="icon-after"
                  ></ecl-icon>
                </ecl-link> : ''
              }
              </div>
              <div
                class="ecl-gallery__detail-description"
                data-ecl-gallery-overlay-description
              >
              </div>
              <div
                class="ecl-gallery__detail-meta"
                data-ecl-gallery-overlay-meta
              >
              </div>    
              <div class="ecl-gallery__pager">
                <ecl-button
                  theme={this.theme}
                  style-class={`ecl-gallery__slider-previous sc-ecl-gallery-${this.theme}`}
                  data-ecl-gallery-overlay-previous
                  variant="ghost"
                  type="button"
                  hide-label
                >
                  <ecl-icon
                    theme={this.theme}
                    style-class={`sc-ecl-gallery-${this.theme}`}
                    slot="icon-before"
                    size="s"
                    transform="rotate-270"
                    icon="corner-arrow"
                  ></ecl-icon>
                  {this.prevLabel}
                </ecl-button>
                <div class="ecl-gallery__detail-counter">
                  <span data-ecl-gallery-overlay-counter-current>0</span>
                    {` ${this.counterSeparator} `}
                  <span data-ecl-gallery-overlay-counter-max>0</span>
                </div>
                <ecl-button
                  theme={this.theme}
                  style-class={`ecl-gallery__slider-next sc-ecl-gallery-${this.theme}`}
                  data-ecl-gallery-overlay-next
                  variant="ghost"
                  type="button"
                  hide-label
                >
                  {this.nextLabel}
                  <ecl-icon
                    theme={this.theme}
                    style-class={`sc-ecl-gallery-${this.theme}`}
                    slot="icon-after"
                    size="s"
                    transform="rotate-90"
                    icon="corner-arrow"
                  ></ecl-icon>
                </ecl-button>
              </div>       
            </div>
          </footer>
        </dialog>
      </section>
    );
  }
}
