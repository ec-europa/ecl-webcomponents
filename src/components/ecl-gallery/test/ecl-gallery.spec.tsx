import { newSpecPage } from '@stencil/core/testing';
import { EclGallery } from '../ecl-gallery';
import { EclGalleryItem } from '../ecl-gallery-item';
import { EclButton } from '../../ecl-button/ecl-button';
import { EclIcon } from '../../ecl-icon/ecl-icon';

describe('ecl-gallery', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclGallery, EclGalleryItem, EclButton, EclIcon],
      html: `<ecl-gallery
        ecl-script
        counter-label="Media files in this gallery"
        view-all-label="View all"
        counter-separator="of"
        next-label="Next"
        prev-label="Previous"
        close-label="Close"
        download-label="Download"
        share-label="Share"
      >
        <ecl-gallery-item
          theme="ec"
          description="The EU in brief, institutions and bodies, countries, symbols, history, facts and figures"
          meta="Copyright, Author, Licence for image 1"
          media-share-path="/example.html"
          media-href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
          image-alt="Image 1"
        >    
        </ecl-gallery-item>
        <ecl-gallery-item
          theme="ec"
          type="video"
          description="New digital strategy"
          meta="Copyright, Author, Licence for embedded media"
          media-href="https://inno-ecl.s3.amazonaws.com/media/examples/example-external-video.jpg"
          image-alt="New digital strategy"
          media-iframe-href="https://www.youtube.com/embed/fgi-GSCB6ho"
        >    
        </ecl-gallery-item>
        <ecl-gallery-item
          theme="ec"
          description="Information on agriculture, business, culture, health, etc."
          meta="Copyright, Author, Licence for image 2"
          media-share-path="/example.html"
          media-href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image2.jpg"
          image-alt="Image 2"
        ></ecl-gallery-item>
        <ecl-gallery-item
          theme="ec"
          description="Living, working, travelling in the EU"
          meta="Copyright, Author, Licence for image 3"
          media-share-path="/example.html"
          media-href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image3.jpg"
          image-alt="Image 3"
        ></ecl-gallery-item>
        <ecl-gallery-item
          theme="ec"
          description="Information on taxes, customs, importing and exporting goods, financial support for businesses"
          meta="Copyright, Author, Licence for image 4"
          media-share-path="/example.html"
          media-href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image4.jpg"
          image-alt="Image 3"
        ></ecl-gallery-item>
        <ecl-gallery-item
          theme="ec"
          media-share-path="/example.html"
          meta="Copyright, Author, Licence for video"
          description="EU law"
          type="html-video"
        >
          <video
            class="ecl-gallery__image sc-ecl-gallery-ec"
            poster="https://inno-ecl.s3.amazonaws.com/media/examples/example-image5.jpg"
          >
            <source
              src="https://inno-ecl.s3.amazonaws.com/media/videos/big_buck_bunny.mp4"
              type="video/mp4"
            />
            <source
              src="https://inno-ecl.s3.amazonaws.com/media/videos/big_buck_bunny.webm"
              type="video/webm"
            />
            <track
              kind="captions"
              label="English"
              src="/captions/bunny-en.vtt"
              srclang="en"
            />
            <track
              kind="captions"
              label="franÃ§ais"
              src="/captions/bunny-fr.vtt"
              srclang="fr"
            />
          </video>
        </ecl-gallery-item>
        <ecl-gallery-item
          theme="ec"
          meta="Copyright, Author, Licence for image 6"
          description="Find official documents, publications, statistics, open data and more resources"
          media-share-path="/example.html"
          media-href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image6.jpg"
          image-alt="Image 6"
        ></ecl-gallery-item>
        <ecl-gallery-item
          theme="ec"
          meta="Copyright, Author, Licence for image 7"
          description="Teaching material about the EU, including books and maps"
          media-share-path="/example.html"
          media-href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image7.jpg"
          image-alt="Image 7"
        ></ecl-gallery-item>
        <ecl-gallery-item
          theme="ec"
          meta="Copyright, Author, Licence for image 8"
          description="Teaching material about the EU, including books and maps"
          media-share-path="/example.html"
          media-iframe-href="https://audiovisual.ec.europa.eu/embed/index.html?ref=I-223223&lg=EN"
          media-href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image8.jpg"
          image-alt="Image 8"
        ></ecl-gallery-item>
        <ecl-gallery-item
          theme="ec"
          meta="Copyright, Author, Licence for image 9"
          description="About the EU"
          media-share-path="/example.html"
          media-href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image9.jpg"
          image-alt="Image 9"
        ></ecl-gallery-item>
        <ecl-gallery-item
          theme="ec"
          meta="Copyright, Author, Licence for image 10"
          description="EU by topic"
          media-share-path="/example.html"
          media-href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image10.jpg"
          image-alt="Image 10"
        ></ecl-gallery-item>
      </ecl-gallery>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
