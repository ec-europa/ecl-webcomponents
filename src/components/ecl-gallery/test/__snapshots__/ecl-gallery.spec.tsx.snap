// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-gallery renders 1`] = `
<ecl-gallery close-label="Close" counter-label="Media files in this gallery" counter-separator="of" download-label="Download" ecl-script="" next-label="Next" prev-label="Previous" share-label="Share" theme="ec" view-all-label="View all">
  <!---->
  <section class="ecl-gallery" data-ecl-gallery data-ecl-gallery-visible-items="8">
    <ul class="ecl-gallery__list">
      <li class="ecl-gallery__item sc-ecl-gallery-ec">
        <a class="ecl-gallery__item-link sc-ecl-gallery-ec" data-ecl-gallery-item data-ecl-gallery-item-share="/example.html" href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg">
          <figure class="ecl-gallery__image-container sc-ecl-gallery-ec">
            <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg" image-alt="Image 1" img-class="ecl-gallery__image sc-ecl-gallery-ec" styleclass="ecl-gallery__picture sc-ecl-gallery-ec"></ecl-picture>
            <figcaption class="ecl-gallery__description sc-ecl-gallery-ec" data-ecl-gallery-description>
              The EU in brief, institutions and bodies, countries, symbols, history, facts and figures
              <span class="ecl-gallery__meta sc-ecl-gallery-ec" data-ecl-gallery-meta>
                Copyright, Author, Licence for image 1
              </span>
            </figcaption>
          </figure>
        </a>
      </li>
      <li class="ecl-gallery__item sc-ecl-gallery-ec">
        <a class="ecl-gallery__item-link sc-ecl-gallery-ec" data-ecl-gallery-item data-ecl-gallery-item-embed-src="https://www.youtube.com/embed/fgi-GSCB6ho" href="https://www.youtube.com/embed/fgi-GSCB6ho">
          <figure class="ecl-gallery__image-container sc-ecl-gallery-ec">
            <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example.html-external-video.jpg" image-alt="New digital strategy" img-class="ecl-gallery__image sc-ecl-gallery-ec" styleclass="ecl-gallery__picture sc-ecl-gallery-ec"></ecl-picture>
            <div class="ecl-gallery__item-icon-wrapper sc-ecl-gallery-ec">
              <ecl-icon style-class="ecl-gallery__item-icon sc-ecl-gallery-ec">
                <svg class="ecl-gallery__item-icon ecl-icon ecl-icon--inverted ecl-icon--l sc-ecl-gallery-ec">
                  <use xlink:href="/build/images/ec/icons.svg#play-filled"></use>
                </svg>
              </ecl-icon>
            </div>
            <figcaption class="ecl-gallery__description sc-ecl-gallery-ec" data-ecl-gallery-description>
              New digital strategy
              <span class="ecl-gallery__meta sc-ecl-gallery-ec" data-ecl-gallery-meta>
                Copyright, Author, Licence for embedded media
              </span>
            </figcaption>
          </figure>
        </a>
      </li>
      <li class="ecl-gallery__item sc-ecl-gallery-ec">
        <a class="ecl-gallery__item-link sc-ecl-gallery-ec" data-ecl-gallery-item data-ecl-gallery-item-share="/example.html" href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image2.jpg">
          <figure class="ecl-gallery__image-container sc-ecl-gallery-ec">
            <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image2.jpg" image-alt="Image 2" img-class="ecl-gallery__image sc-ecl-gallery-ec" styleclass="ecl-gallery__picture sc-ecl-gallery-ec"></ecl-picture>
            <figcaption class="ecl-gallery__description sc-ecl-gallery-ec" data-ecl-gallery-description>
              Information on agriculture, business, culture, health, etc.
              <span class="ecl-gallery__meta sc-ecl-gallery-ec" data-ecl-gallery-meta>
                Copyright, Author, Licence for image 2
              </span>
            </figcaption>
          </figure>
        </a>
      </li>
      <li class="ecl-gallery__item sc-ecl-gallery-ec">
        <a class="ecl-gallery__item-link sc-ecl-gallery-ec" data-ecl-gallery-item data-ecl-gallery-item-share="/example.html" href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image3.jpg">
          <figure class="ecl-gallery__image-container sc-ecl-gallery-ec">
            <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image3.jpg" image-alt="Image 3" img-class="ecl-gallery__image sc-ecl-gallery-ec" styleclass="ecl-gallery__picture sc-ecl-gallery-ec"></ecl-picture>
            <figcaption class="ecl-gallery__description sc-ecl-gallery-ec" data-ecl-gallery-description>
              Living, working, travelling in the EU
              <span class="ecl-gallery__meta sc-ecl-gallery-ec" data-ecl-gallery-meta>
                Copyright, Author, Licence for image 3
              </span>
            </figcaption>
          </figure>
        </a>
      </li>
      <li class="ecl-gallery__item sc-ecl-gallery-ec">
        <a class="ecl-gallery__item-link sc-ecl-gallery-ec" data-ecl-gallery-item data-ecl-gallery-item-share="/example.html" href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image4.jpg">
          <figure class="ecl-gallery__image-container sc-ecl-gallery-ec">
            <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image4.jpg" image-alt="Image 3" img-class="ecl-gallery__image sc-ecl-gallery-ec" styleclass="ecl-gallery__picture sc-ecl-gallery-ec"></ecl-picture>
            <figcaption class="ecl-gallery__description sc-ecl-gallery-ec" data-ecl-gallery-description>
              Information on taxes, customs, importing and exporting goods, financial support for businesses
              <span class="ecl-gallery__meta sc-ecl-gallery-ec" data-ecl-gallery-meta>
                Copyright, Author, Licence for image 4
              </span>
            </figcaption>
          </figure>
        </a>
      </li>
      <li class="ecl-gallery__item sc-ecl-gallery-ec">
        <a class="ecl-gallery__item-link sc-ecl-gallery-ec" data-ecl-gallery-item data-ecl-gallery-item-share="/example.html">
          <figure class="ecl-gallery__image-container sc-ecl-gallery-ec">
            <video class="ecl-gallery__image sc-ecl-gallery-ec" poster="https://inno-ecl.s3.amazonaws.com/media/examples/example-image6.jpg" slot="video">
              <source src="https://inno-ecl.s3.amazonaws.com/media/videos/big_buck_bunny.mp4" type="video/mp4">
              <source src="https://inno-ecl.s3.amazonaws.com/media/videos/big_buck_bunny.webm" type="video/webm">
              <track kind="captions" label="English" src="/captions/bunny-en.vtt" srclang="en">
              <track kind="captions" label="franÃ§ais" src="/captions/bunny-fr.vtt" srclang="fr">
            </video>
            <div class="ecl-gallery__item-icon-wrapper sc-ecl-gallery-ec">
              <ecl-icon style-class="ecl-gallery__item-icon sc-ecl-gallery-ec">
                <svg class="ecl-gallery__item-icon ecl-icon ecl-icon--inverted ecl-icon--l sc-ecl-gallery-ec">
                  <use xlink:href="/build/images/ec/icons.svg#play-filled"></use>
                </svg>
              </ecl-icon>
            </div>
            <figcaption class="ecl-gallery__description sc-ecl-gallery-ec" data-ecl-gallery-description>
              EU law
              <span class="ecl-gallery__meta sc-ecl-gallery-ec" data-ecl-gallery-meta>
                Copyright, Author, Licence for video
              </span>
            </figcaption>
          </figure>
        </a>
      </li>
      <li class="ecl-gallery__item sc-ecl-gallery-ec">
        <a class="ecl-gallery__item-link sc-ecl-gallery-ec" data-ecl-gallery-item data-ecl-gallery-item-share="/example.html" href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image7.jpg">
          <figure class="ecl-gallery__image-container sc-ecl-gallery-ec">
            <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image7.jpg" image-alt="Image 6" img-class="ecl-gallery__image sc-ecl-gallery-ec" styleclass="ecl-gallery__picture sc-ecl-gallery-ec"></ecl-picture>
            <figcaption class="ecl-gallery__description sc-ecl-gallery-ec" data-ecl-gallery-description>
              Find official documents, publications, statistics, open data and more resources
              <span class="ecl-gallery__meta sc-ecl-gallery-ec" data-ecl-gallery-meta>
                Copyright, Author, Licence for image 6
              </span>
            </figcaption>
          </figure>
        </a>
      </li>
      <li class="ecl-gallery__item sc-ecl-gallery-ec">
        <a class="ecl-gallery__item-link sc-ecl-gallery-ec" data-ecl-gallery-item data-ecl-gallery-item-share="/example.html" href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image8.jpg">
          <figure class="ecl-gallery__image-container sc-ecl-gallery-ec">
            <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image8.jpg" image-alt="Image 7" img-class="ecl-gallery__image sc-ecl-gallery-ec" styleclass="ecl-gallery__picture sc-ecl-gallery-ec"></ecl-picture>
            <figcaption class="ecl-gallery__description sc-ecl-gallery-ec" data-ecl-gallery-description>
              Teaching material about the EU, including books and maps
              <span class="ecl-gallery__meta sc-ecl-gallery-ec" data-ecl-gallery-meta>
                Copyright, Author, Licence for image 7
              </span>
            </figcaption>
          </figure>
        </a>
      </li>
      <li class="ecl-gallery__item sc-ecl-gallery-ec">
        <a class="ecl-gallery__item-link sc-ecl-gallery-ec" data-ecl-gallery-item data-ecl-gallery-item-embed-src="https://audiovisual.ec.europa.eu/embed/index.html?ref=I-223223&amp;lg=EN" data-ecl-gallery-item-share="/example.html" href="https://audiovisual.ec.europa.eu/embed/index.html?ref=I-223223&amp;lg=EN">
          <figure class="ecl-gallery__image-container sc-ecl-gallery-ec">
            <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image9.jpg" image-alt="Image 8" img-class="ecl-gallery__image sc-ecl-gallery-ec" styleclass="ecl-gallery__picture sc-ecl-gallery-ec"></ecl-picture>
            <figcaption class="ecl-gallery__description sc-ecl-gallery-ec" data-ecl-gallery-description>
              Teaching material about the EU, including books and maps
              <span class="ecl-gallery__meta sc-ecl-gallery-ec" data-ecl-gallery-meta>
                Copyright, Author, Licence for image 8
              </span>
            </figcaption>
          </figure>
        </a>
      </li>
      <li class="ecl-gallery__item sc-ecl-gallery-ec">
        <a class="ecl-gallery__item-link sc-ecl-gallery-ec" data-ecl-gallery-item data-ecl-gallery-item-share="/example.html" href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg">
          <figure class="ecl-gallery__image-container sc-ecl-gallery-ec">
            <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg" image-alt="Image 9" img-class="ecl-gallery__image sc-ecl-gallery-ec" styleclass="ecl-gallery__picture sc-ecl-gallery-ec"></ecl-picture>
            <figcaption class="ecl-gallery__description sc-ecl-gallery-ec" data-ecl-gallery-description>
              About the EU
              <span class="ecl-gallery__meta sc-ecl-gallery-ec" data-ecl-gallery-meta>
                Copyright, Author, Licence for image 9
              </span>
            </figcaption>
          </figure>
        </a>
      </li>
      <li class="ecl-gallery__item sc-ecl-gallery-ec">
        <a class="ecl-gallery__item-link sc-ecl-gallery-ec" data-ecl-gallery-item data-ecl-gallery-item-share="/example.html" href="https://inno-ecl.s3.amazonaws.com/media/examples/example-image10.jpg">
          <figure class="ecl-gallery__image-container sc-ecl-gallery-ec">
            <ecl-picture image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image10.jpg" image-alt="Image 10" img-class="ecl-gallery__image sc-ecl-gallery-ec" styleclass="ecl-gallery__picture sc-ecl-gallery-ec"></ecl-picture>
            <figcaption class="ecl-gallery__description sc-ecl-gallery-ec" data-ecl-gallery-description>
              EU by topic
              <span class="ecl-gallery__meta sc-ecl-gallery-ec" data-ecl-gallery-meta>
                Copyright, Author, Licence for image 10
              </span>
            </figcaption>
          </figure>
        </a>
      </li>
    </ul>
    <div class="ecl-gallery__footer">
      <div class="ecl-gallery__info">
        <strong class="ecl-gallery__info-total" data-ecl-gallery-count>
          0
        </strong>
        Media files in this gallery,
        <ecl-button data-ecl-gallery-all data-ecl-gallery-collapsed-label="View all">
          <!---->
          <button class="ecl-button ecl-button--ghost ecl-gallery__view-all sc-ecl-gallery-ec" data-ecl-gallery-all="eclGalleryAll" data-ecl-gallery-collapsed-label="View all" type="submit">
            <span class="ecl-button__container">
              <span class="ecl-button__label">
                View all
              </span>
            </span>
          </button>
        </ecl-button>
      </div>
      <hr class="ecl-gallery__footer-divider">
    </div>
    <dialog class="ecl-gallery__overlay" data-ecl-gallery-overlay>
      <header class="ecl-gallery__close" data-ecl-gallery-overlay-header>
        <ecl-button data-ecl-gallery-close style-class="ecl-gallery__close-button sc-ecl-gallery-ec">
          <!---->
          <button class="ecl-button ecl-button--ghost ecl-gallery__close-button sc-ecl-gallery-ec" data-ecl-gallery-close="eclGalleryClose" type="submit">
            <span class="ecl-button__container">
              <span class="ecl-button__label">
                Close
              </span>
              <ecl-icon slot="icon-after" style-class="sc-ecl-gallery-ec">
                <svg class="ecl-button__icon ecl-button__icon--after ecl-icon ecl-icon--s sc-ecl-button-ec sc-ecl-gallery-ec">
                  <use xlink:href="/build/images/ec/icons.svg#close-filled"></use>
                </svg>
              </ecl-icon>
            </span>
          </button>
        </ecl-button>
      </header>
      <section class="ecl-gallery__slider">
        <div class="ecl-gallery__slider-media-container" data-ecl-gallery-overlay-media></div>
      </section>
      <footer class="ecl-gallery__detail" data-ecl-gallery-overlay-footer>
        <div class="ecl-container">
          <div class="ecl-gallery__detail-actions">
            <ecl-link data-ecl-gallery-overlay-download path="" style-class="ecl-gallery__download sc-ecl-gallery-ec" target="blank" theme="ec" variant="standalone">
              <ecl-icon slot="icon-after">
                <svg class="ecl-icon ecl-icon--fluid">
                  <use xlink:href="/build/images/ec/icons.svg#fullscreen"></use>
                </svg>
              </ecl-icon>
            </ecl-link>
            <ecl-link data-ecl-gallery-overlay-share path="" style-class="ecl-gallery__share sc-ecl-gallery-ec" theme="ec" variant="standalone">
              Share
              <ecl-icon slot="icon-after">
                <svg class="ecl-icon ecl-icon--fluid">
                  <use xlink:href="/build/images/ec/icons.svg#share"></use>
                </svg>
              </ecl-icon>
            </ecl-link>
          </div>
          <div class="ecl-gallery__detail-description" data-ecl-gallery-overlay-description></div>
          <div class="ecl-gallery__detail-meta" data-ecl-gallery-overlay-meta></div>
          <div class="ecl-gallery__pager">
            <ecl-button data-ecl-gallery-overlay-previous hide-label="" style-class="ecl-gallery__slider-previous sc-ecl-gallery-ec">
              <!---->
              <button class="ecl-button ecl-button--ghost ecl-button--icon-only ecl-gallery__slider-previous sc-ecl-gallery-ec" data-ecl-gallery-overlay-previous="eclGalleryOverlayPrevious" type="button">
                <span class="ecl-button__container">
                  <ecl-icon slot="icon-before" style-class="sc-ecl-gallery-ec">
                    <svg class="ecl-button__icon ecl-icon ecl-icon--rotate-270 ecl-icon--s sc-ecl-button-ec sc-ecl-gallery-ec">
                      <use xlink:href="/build/images/ec/icons.svg#corner-arrow"></use>
                    </svg>
                  </ecl-icon>
                  <span class="ecl-u-sr-only" data-ecl-label>
                    Previous
                  </span>
                </span>
              </button>
            </ecl-button>
            <div class="ecl-gallery__detail-counter">
              <span data-ecl-gallery-overlay-counter-current>
                0
              </span>
              of
              <span data-ecl-gallery-overlay-counter-max>
                0
              </span>
            </div>
            <ecl-button data-ecl-gallery-overlay-next hide-label="" style-class="ecl-gallery__slider-next sc-ecl-gallery-ec">
              <!---->
              <button class="ecl-button ecl-button--ghost ecl-button--icon-only ecl-gallery__slider-next sc-ecl-gallery-ec" data-ecl-gallery-overlay-next="eclGalleryOverlayNext" type="button">
                <span class="ecl-button__container">
                  <span class="ecl-u-sr-only" data-ecl-label>
                    Next
                  </span>
                  <ecl-icon slot="icon-after" style-class="sc-ecl-gallery-ec">
                    <svg class="ecl-button__icon ecl-icon ecl-icon--rotate-90 ecl-icon--s sc-ecl-button-ec sc-ecl-gallery-ec">
                      <use xlink:href="/build/images/ec/icons.svg#corner-arrow"></use>
                    </svg>
                  </ecl-icon>
                </span>
              </button>
            </ecl-button>
          </div>
        </div>
      </footer>
    </dialog>
  </section>
</ecl-gallery>
`;
